rename

NadiaArroyo25 opened the pull request:
Rename my model and table in my database to user, and also the column in the medical records table

NadiaArroyo25 committed:
rename

Rename my model and table in my database to user, and also the column in the medical records table

Ligator commented on app/views/shared/_header.html.erb:
```diff
@@ -9,21 +9,18 @@
+        <% if user_signed_in? %>
         <%= link_to "Mis pacientes", patients_path, class: "nav-item nav-link" %>
-        <%= button_to "Cerrar Sesión", destroy_usuario_session_path, method: :delete, class: "nav-item nav-link btn btn-outline-secondary btn-sm" %>
+        <%= button_to "Cerrar Sesión", destroy_user_session_path, method: :delete, class: "nav-item nav-link btn btn-outline-secondary btn-sm" %>
```
Indentation.

AlfredoJA commented on db/migrate/20240210054714_devise_create_usuarios.rb:
```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
-class DeviseCreateUsuarios < ActiveRecord::Migration[7.1]
+class DeviseCreateUsers < ActiveRecord::Migration[7.1]
```
For old migrations is not necessary did this change. So revert changes for this file and below.

NadiaArroyo25 replied:
Just the content right?

Ligator replied:
No, the entire file should be as it was.

NadiaArroyo25 replied:
ok!

Ligator commented on db/migrate/20240210054714_devise_create_usuarios.rb:
```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
-class DeviseCreateUsuarios < ActiveRecord::Migration[7.1]
+class DeviseCreateUsers < ActiveRecord::Migration[7.1]
```
Previous migrations must no be modified, ever!.
The reason is that, when you want to clone your repository in another computer and run the migrations, a `users` table will be created in this migration and when you run the next migration to rename the table from `usuarios` to `users` it won't find the `usuarios` table.
So, please leave this migration as it was.

NadiaArroyo25 replied:
D:! oh no!

Ligator commented on db/migrate/20240507004300_add_user_id_to_medical_records.rb:
```diff
@@ -1,9 +1,9 @@
+    add_reference :medical_records, :user, null: true, foreign_key: true, default: user.id
 
-    change_column_default :medical_records, :usuario_id, nil
+    change_column_default :medical_records, :user_id, nil
```
Same here.

NadiaArroyo25 replied:
Ok, I understand but only the previous migrations, right? I mean only the content of the files?

Ligator replied:
You may have a problem running this migration anyway because if you look for the first "Usuario", the migration won't find this model even when the table `usuarios` is there, this is because your code now has the model `User`.
On the other hand, if you change your query to look for the first `User`, the migration won't find a table `users` that matches the model, this is because at this point the `users` table doesn't exists.
So, given this case scenario, you need to assign the ID number directly rather than look for the record and hope it works whenever you run it. (Yes, I am suggesting to go back as how it was at the beginning)

NadiaArroyo25 committed:
correction of past migrations

Ligator commented on db/schema.rb:
```diff
@@ -105,22 +105,22 @@
-    t.index ["email"], name: "index_usuarios_on_email", unique: true
-    t.index ["reset_password_token"], name: "index_usuarios_on_reset_password_token", unique: true
+    t.index ["email"], name: "index_users_on_email", unique: true
+    t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
```
Did you edit manually the `schema.rb` file?
This file should be modified automatically by running the migrations. That said, I didn't see a migration where you changed the indexes. Were these indexes affected by your current migrations only?

NadiaArroyo25 replied:
No, I modified them because I wasn't sure about rolling back, but the migrations I did not change it, in fact the change was after the migrations had already been done manually

NadiaArroyo25 replied:
The modifications that I made manually, can I revert them manually? Or will it cause a problem like in this case?

NadiaArroyo25 replied:
I will do it again

Ligator replied:
Just run your migrations again and keep the schema as it will be.
Then, if there is still any index with `usuario`, let us know, so we can create a migration to fix it.

NadiaArroyo25 replied:
I already modified it, that's it and it works :D

NadiaArroyo25 committed:
Revert "correction of past migrations"

This reverts commit 81adc235ae6621f81d4700654bc7d6a30471ff56.

NadiaArroyo25 committed:
delete manual changes after suggestion

delete the changes you made manually because you had not reverted the migrations

NadiaArroyo25 committed:
this file to

NadiaArroyo25 committed:
Revert "rename"

This reverts commit 0d778ceca0b0a9af265b59c43d5a2cbeba308f1f.

NadiaArroyo25 committed:
new migrations and changes

I redid the migrations and changes, without touching past migrations or anything in the database

NadiaArroyo25 committed:
space
